src/pconn.cc

```diff
@@ -582,22 +584,21 @@ PconnModule::remove(PconnPool *aPool)
 }
 
 void
-PconnModule::dump(StoreEntry *e)
+PconnModule::dump(std::ostream &yaml)
 {
-    typedef Pools::const_iterator PCI;
-    int i = 0; // TODO: Why number pools if they all have names?
-    for (PCI p = pools.begin(); p != pools.end(); ++p, ++i) {
+    for (const auto &p: pools) {
         // TODO: Let each pool dump itself the way it wants to.
-        storeAppendPrintf(e, "\n Pool %d Stats\n", i);
-        (*p)->dumpHist(e);
-        storeAppendPrintf(e, "\n Pool %d Hash Table\n",i);
-        (*p)->dumpHash(e);
+        yaml << "pool " << p->description() << "\n" <<
+             "  open connections: " << p->count() << "\n";
+        p->dumpHash(yaml);
+        p->dumpHist(yaml);
```

Comment:
Please move code inside this loop into PconnPool::dump(). This will remove the need to add description() and will allow us to hide dumpHash() and dumpHist() as private methods that do not need to be STUBbed. Since we are changing every line in this loop except for the TODO comment that asks us to make these changes, I think it is time to make these changes.

Reply:
done